client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java

```diff
@@ -0,0 +1,302 @@
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.appcompat.view.ActionMode;
+import androidx.databinding.DataBindingUtil;
+import androidx.preference.PreferenceManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import dagger.android.support.DaggerAppCompatActivity;
+import io.mosip.registration.app.R;
+import io.mosip.registration.app.databinding.PacketListActivityBinding;
+import io.mosip.registration.app.util.NetworkUtil;
+import io.mosip.registration.app.util.PacketListActivityHelper;
+import io.mosip.registration.app.viewmodel.RegistrationPacketListAdapter;
+import io.mosip.registration.app.viewmodel.RegistrationPacketViewModel;
+import io.mosip.registration.app.viewmodel.model.RegistrationPacketModel;
+import io.mosip.registration.clientmanager.constant.AuditEvent;
+import io.mosip.registration.clientmanager.constant.AuditReferenceIdTypes;
+import io.mosip.registration.clientmanager.constant.Components;
+import io.mosip.registration.clientmanager.constant.RegistrationConstants;
+import io.mosip.registration.clientmanager.repository.GlobalParamRepository;
+import io.mosip.registration.clientmanager.spi.AuditManagerService;
+import io.mosip.registration.clientmanager.spi.PacketService;
+import io.mosip.registration.clientmanager.spi.PacketUploadService;
+import io.mosip.registration.clientmanager.util.DateUtil;
+
+public class PacketListActivity extends DaggerAppCompatActivity {
+
+    private static final String TAG = PacketListActivity.class.getSimpleName();
+
+    private static final int PROGRESS_RESET = 0;
+    private static final int PROGRESS_SYNC_STARTED = 20;
+    private static final int PROGRESS_SYNC_COMPLETED = 40;
+    private static final int PROGRESS_UPLOAD_STARTED = 60;
+    private static final int PROGRESS_UPLOAD_COMPLETED = 100;
+
+    @Inject
+    PacketService packetService;
+
+    @Inject
+    PacketUploadService packetUploadService;
+
+    @Inject
+    DateUtil dateUtil;
+
+    @Inject
+    AuditManagerService auditManagerService;
+
+    RegistrationPacketViewModel registrationPacketViewModel;
+
+    PacketListActivityBinding bi;
+    List<RegistrationPacketModel> list;
+    RegistrationPacketListAdapter adapter;
+    ActionMode actionMode;
+    ActionCallback actionCallback;
+    SwipeRefreshLayout swipeRefreshLayout;
+    Boolean uploadOverWifiOnly;
+    SharedPreferences sharedPreferences;
+    NetworkUtil networkUtil;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        bi = DataBindingUtil.setContentView(this, R.layout.packet_list_activity);
+        networkUtil = new NetworkUtil(this);
+
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this /* Activity context */);
+        uploadOverWifiOnly = sharedPreferences.getBoolean("wifi_only", false);
```

Comment:
Is the idea here to take this as a setting and to store it in the sharedPreferences?

Reply:
this is a shared preferences setting. loading the value into uploadOverWifiOnly variable on the activity start and resume